### ISSUE TITLE
DataTables warning / no report

### ISSUE THREAD

**Issue by stueja** [meeting-room-booking-system/mrbs-code#3723]
**Describe the problem you're facing/question you have**
Yesterday, using a docker container which I built without the `--no-cache` option, and therefore version 1.11.4, I spun up a test installation. I added three more columns within `mrbs_entry` and  `mrbs_repeat`. I gave them talkative names using `$vocab_override`, made them mandatory using `$is_mandatory_field`. I successfully ran an html report.

Today, I spun up a totally fresh container with the current version (`/refs/heads/main.tar.gz`). Added the columns as above. No problem so far. Users can be listed and edited, rooms can be listed and edited, but `Report -- Create Report` brings up this message: `DataTables warning: table id=report_table - Invalid JSON response. For more information about this error, please see http://datatables.net/tn/1`.

The database name is `mrbs-test`, the prefix is `mrbs_`.

**Browser details (please complete the following information):**
 - OS: macOS 14.6.1
 - Browser Safari 17.6 and Firefox 129.0.1

**Server details (please complete the following information):**
 - MRBS version 1.11.5
 - Web server: nginx/1.22.1
 - OS: Linux d446d0e82d99 5.15.0-113-generic 123-Ubuntu
 - PHP version: 8.1.20
 - Database used: mariadb 11.4.3
 - database schema:	82
 - version of local database schema	1

**Additional context**
Downloading the csv instead of running the html report downloads a csv containing the html source code of mrbs.

Not sure on how to go on from here. Where can I start?

**Comment by campbell-m** [comment]
Have a look in your PHP error log to see the details of the error message. I suspect there was a PHP error which led to a corrupt JSON file.

Or else set `$debug = true;` in your config file and then look in your browser's developer tools at the response from report.php after you have acknowledged the datatables error message.

**Comment by stueja** [comment]
Thanks alot for the guidance.

The console reports
```
Source-Map-Fehler: request failed with status 404
Ressourcen-Adresse: https://sub.domain.tld/jquery/datatables/datatables.js?v=1723656502
Source-Map-Adresse: pdfmake.js.map
```

And the xhr response is
```
Uncaught exception 'TypeError' in /var/www/html/report.php at line 550
MRBS\escape(): Argument #1 ($string) must be of type string, int given, called in /var/www/html/report.php on line 1026
#0 /var/www/html/report.php(1026): MRBS\escape()
#1 /var/www/html/report.php(1945): MRBS\report_row()
#2 {main}

MRBS GET: Array
(
)
MRBS POST: Array
(
    [MAX_FILE_SIZE] => 2097152
    [csrf_token] => 3ca70c4a49ba3055b3030f43aedc4cc784d554caa1a74a6b06a8b67a3b2f6de3
    [phase] => 2
    [from_date] => 2024-07-01
    [to_date] => 2024-10-14
    [areamatch] =>
    [roommatch] =>
    [namematch] =>
    [descrmatch] =>
    [creatormatch] =>
    [match_miteinkommen] =>
    [match_ohneeinkommen] =>
    [match_arbeitsstunden] =>
    [output] => 0
    [output_format] => 0
    [sortby] => r
    [sumby] => d
    [datatable] => 1
)
MRBS SESSION: Array
(
    [last_page] => /report.php
    [this_page] => /report.php
    [csrf_token] => 3ca70c4a49ba3055b3030f43aedc4cc784d554caa1a74a6b06a8b67a3b2f6de3
    [user] => MRBS\User Object
        (
            [username] => username
            [display_name] => User Name
            [email] => user.name@email.address
            [level] => 2
            [data:protected] => Array
                (
                    [id] => 1
                    [password_hash] => $2y$10$TVF7aK/wkKhmwOPhWO/Qze0/Bda4nlM7cVO3UmI5LwvIM7ns.1ZX2
                    [timestamp] => 2024-02-19 20:25:47
                    [last_login] => 1723746860
                    [reset_key_hash] =>
                    [reset_key_expiry] => 0
                )

        )
```

**Comment by campbell-m** [comment]
Now fixed in [5ad7ed2](https://github.com/meeting-room-booking-system/mrbs-code/commit/5ad7ed21bf7079ab62f796c5c3762e420dd5ca08).

**Comment by stueja** [comment]
What should I say. Tested, works as expected, thank you for the fast fix!
